modules/cudf/src/dataframe/print.ts

```diff
@@ -0,0 +1,150 @@
+// Copyright (c) 2021, NVIDIA CORPORATION.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {DataFrame} from '../data_frame';
+import {Series} from '../series';
+import {Int32, Utf8String} from '../types/dtypes';
+import {TypeMap} from '../types/mappings';
+
+export type DisplayOptions = {
+  /**
+   * The maximum number of columns to display before truncation. Truncation may also occur
+   * if the total printed width in characters exceeds `widht
+   *
+   */
+  maxColumns?: number,
+
+  /**
+   * The maximum width in characters of a column when printing a DataFram. When the column *
+   * overflows, a "..." placeholder is embedded in the output. A value of zero means unlimited.
+   * (default: 50)
+   */
+  maxColWidth?: number,
+
+  /**
+   * The maximum number of rows to output when printing a DataFrame.  A value of zero means
+   * unlimited. (default: 60)
+   */
+  maxRows?: number,
+
+  /**
+   * Width of the display in characters. (default: 80)
+   */
+  width?: number,
+
+}
+
+export class DataFrameFormatter<T extends TypeMap> {
+  private obj: DataFrame;
+  private maxColumns: number;
+  private maxColWidth: number;
+  private maxRows: number;
+  private width: number;
+  private htrunc: boolean;
+  private vtrunc: boolean;
+
+  constructor(options: DisplayOptions, obj: DataFrame<T>) {
+    this.maxColumns  = options.maxColumns ?? 20;
+    this.maxColWidth = options.maxColWidth ?? 50;
+    this.maxRows     = options.maxRows ?? 60;
+    this.width       = options.width ?? 80;
+    this.htrunc      = false;
+    this.vtrunc      = false;
+
+    let tmp = this.preprocess(obj);
+    while (this._totalWidth(tmp) > this.width) {
+      this.htrunc = true;
+      const names = [...tmp.names];
+      const N     = Math.ceil((names.length - 1) / 2);
+      tmp         = tmp.drop([names[N]]);
+    }
+    this.obj = tmp;
+  }
+
+  preprocess(obj: DataFrame<T>) {
```

Comment:
If you decide on changing the `obj` param name above.

Reply:
https://github.com/rapidsai/node/pull/297/commits/842a5bd781f3a403a745a759df09936256c70944